### DISCUSSION TITLE
Turtorial for changing the UI?

### DISCUSSION THREAD

**Discussion by TronCarter269** [rzeldent/esp32-smartdisplay#205] [Q&A]
I have been able to get the demo running with some other forum help.  I have also figured out Squareline in order to make a new UI, but can't figure out how to insert my UI into the demo instead of the included one.  Has anyone written such a tutorial?   I tried copying all of the files over and deleting a bunch of stuff from main.cpp.  I am able to compile and upload without error, but I just get the same demo screen, but no touch functionality (probably because of my main.cpp editing).  What is the export and import process?

**Comment by NorthernMan54** [comment]
I'm just as much a beginner as you are, but I found with Squareline, I just needed to set to the 'File-->Project Settings-->UI Files Export Path' to '.../esp32-smartdisplay-demo/src/ui' and hit the export button.  I used the stock main.cpp

**Reply by TronCarter269** [reply]
OK, that worked for a basic screen I made in SquareLine.  I did have to change some things in main.cpp that were looking for things on the old UI.  

The next thing I would like to figure out is how to set the rotation to be landscape instead of portrait.  I tried changing it in Squareline, but it didn't make any change.   I think it happens here is main.cpp, but I don't know what to do:  If I uncomment any of the lines. I get an error that LV_DISP_ROT_90 is undefined.

`    __attribute__((unused)) auto disp = lv_disp_get_default();
    // lv_disp_set_rotation(disp, LV_DISP_ROT_90);
    // lv_disp_set_rotation(disp, LV_DISP_ROT_180);
    // lv_disp_set_rotation(disp, LV_DISP_ROT_270);
`

**Comment by rzeldent** [comment]
There is a lot of information about LVGL and it's use out there. For starters: https://docs.lvgl.io/master/get-started/index.html

Also on youtube you'll find lot of information about LVGL!

It is not easy, that's true but that goes also for embedded programming...
